src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/RegularExpressionAttribute.cs

```diff
@@ -66,11 +66,14 @@ public override bool IsValid(object? value)
                 return true;
             }
 
-            var m = Regex!.Match(stringValue);
+            foreach (ValueMatch m in Regex!.EnumerateMatches(stringValue))
```

Comment:
NIT: since we are only using the first result of the enumerator, would it make sense to instead call MoveNext/Current directly?

Reply:
I've gone back and forth on this one. The generated code is slightly smaller without a loop, but it didn't have a measurable impact and I personally find the foreach easier to read and reason about, so I'd like to leave it.